client/gulpfile.js

```diff
@@ -205,6 +208,64 @@ function buildPlugins(callback, forceRebuild) {
     return callback();
 }
 
+async function installPlugins(callback) {
+    // iterate through install_plugin_build_ids, identify xml files and install dependencies
+    for (const plugin_name of INSTALL_PLUGIN_BUILD_IDS) {
+        const pluginDir = path.join(PATHS.pluginBaseDir, `visualizations/${plugin_name}`);
+        const xmlPath = path.join(pluginDir, `config/${plugin_name}.xml`);
+        // Check if the file exists
+        if (fs.existsSync(xmlPath)) {
+            await installDependenciesFromXML(xmlPath, pluginDir);
+        } else {
+            console.error(`XML file not found: ${xmlPath}`);
+        }
+    }
+    return callback();
+}
+
+// Function to parse the XML and install dependencies
+async function installDependenciesFromXML(xmlPath, pluginDir) {
+    try {
+        const pluginXML = fs.readFileSync(xmlPath);
```

Comment:
I think on the backend you could use macros (which aren't real XML macros). I don't think we do, but that's a potential gotcha ?

Reply:
Good note for the future -- I don't think we even support those macros in viz xml, but eventually this all moves out of gulp into python and at that point we'll be using the same parsing instead of this lighter weight approach.

Reply:
macros will work in visualizations since they're loaded via `load_with_references`, but if the plan is to it in python anyway it's probably fine ?

Reply:
Should have looked before I spoke -- we of course actually do use macros in many visualizations with similar configurations (like the 10 nvd3 viz's, etc.).  Good motivation to merge all this into the backend sooner rather than later.

Reply:
On the bright side the requirements tag didn't mean anything up to now, so this is all fine ... however maybe we should clarify the package type  (`npm` maybe)?

Reply:
Exactly, wouldn't be a problem until someone tries to use a dependency macro and then it'd be pretty obvious it didn't work.

I had a few different versions of how to specify this, let me dig up the options -- happy to tweak it however folks want

Reply:
Eh, the other options either broke things apart more like this:


    <requirement>
        <package>@galaxyproject/msa</package>
        <version>0.2.0</version>
    </requirement>

Which I didn't like -- best to just keep it all in attributes for concise specifications.

Or like this:

<requirements>
    <package>@galaxyproject/msa@0.2.0</package>
</requirements>

Or pushing to the toplevel, like:

<requirement type="node" version="0.2.0" package="@galaxyproject/msa"/>

Without a surrounding requirements/dependencies block.  I actually kind of like the last one since visualizations will have a single external prebuilt package and never multiple packages.